compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -82,16 +82,61 @@ class TaggedShapeAnalyzer final
 
   struct Dim final
   {
-    uint32_t value;
+    int32_t value;
```

Comment:
(NOTE) reshape/shape value can have -1 value.

ref / pytorch reshape said 'A single dimension can be -1' [:link:]( https://pytorch.org/docs/master/generated/torch.reshape.html#torch.reshape:~:text=A%20single%20dimension%20may%20be%20%2D1%2C%20in%20which%20case%20it%E2%80%99s%20inferred%20from%20the%20remaining%20dimensions%20and%20the%20number%20of%20elements%20in%20input.)